cloud/keystone_test.go

```diff
@@ -49,69 +49,136 @@ var _ = Describe("Keystone client", func() {
 		server.Close()
 	})
 
-	Describe("Tenant ID validation", func() {
+	Describe("Keystone validation", func() {
 
-		setupKeystone := func() {
+		expectTokenAuthScopedToDomain := func() {
+			server.AppendHandlers(
+				ghttp.RespondWithJSONEncoded(201, getV3TokensScopedToDomainResponse()),
+			)
+		}
+
+		expectTokenAuthScopedToTenant := func() {
 			server.AppendHandlers(
 				ghttp.RespondWithJSONEncoded(201, getV3TokensScopedToTenantResponse()),
+			)
+		}
+
+		expectTenantList := func() {
+			server.AppendHandlers(
 				ghttp.RespondWithJSONEncoded(200, getV3TenantsResponse()),
 			)
-			setupV3Client()
 		}
 
-		It("Returns false when tenant id is not known", func() {
-			setupKeystone()
+		expectDomainList := func() {
+			server.AppendHandlers(
+				ghttp.RespondWithJSONEncoded(200, getV3DomainsResponse()),
+			)
+		}
 
-			valid, err := client.ValidateTenantID("some-random-id")
-			Expect(valid).To(BeFalse())
-			Expect(err).ToNot(HaveOccurred())
-		})
+		Context("TenantID and DomainID pair validation", func() {
+			It("Valid tenantID within different domain", func() {
```

Comment:
nit: I know you followed the naming from the rest of the tests but imho "_it_ should reject valid tenantID within different domain" sounds better